### PR TITLE
Issue 366: redirect after round apply

### LINKED ISSUE #366: application: after applying to a round, redirect to Project Details page
**User Story**
As a project owner, 
I want to go to the project details page for the project I just submitted
so that I can see my application status 

Note: The "My Applications" section will live on the Project Details page

**Acceptance Criteria**

GIVEN that I just submitted my application 
WHEN I look at the page 
THEN I see that I have been redirected to the Project Details page 

Note: When the application is processing, STAY on application page, do not take user to new page

### PR DESCRIPTION
**Pull request by codenamejason**
Changes:

- [x] Adds redirect to project details page after round application submission
- [x] Adds some basic tests to test the project id and status

### CODE REVIEW

File: client/src/components/grants/Details.tsx [hunk 1]
```diff
@@ -20,16 +22,52 @@ export default function Details({
...
+  const discordLink = (
+    <a className="text-purple-600" href="https://discord.gg/nwYzGuuruJ">
+      Grant Hub Discord
+    </a>
+  );
+
+  const successfullSubmissionBody = (
+    <p className="text-black">
+      Your application will be reviewed by the grant round team and you will
+      receive an email if your project is approved for the grant round. If you
```
Comment (DanieleSalatti) [inline client/src/components/grants/Details.tsx]: This makes me think I will not hear anything back if it's not approved - is that the case, or should we change the copy a little?

@michellema1208 @willsputra

File: client/src/components/grants/Details.tsx [hunk 2]
```diff
@@ -20,16 +22,52 @@ export default function Details({
...
+      have any questions or feedback, feel free to reach us out on the{" "}
+      {discordLink}!
+    </p>
+  );
+
+  // ToDo: make the round name dynamic
+  const successfullSubmissionTitle = (
+    <p className="text-green-800 text-xl">
+      Thank you for applying to Optimism Grant Round!
```
Comment (DanieleSalatti) [inline client/src/components/grants/Details.tsx]: Should this come from the round metadata?

File: client/src/components/rounds/Apply.tsx [hunk 3]
```diff
@@ -90,8 +92,11 @@ function Apply() {
     );
   }
 
+  // todo: navigating to early...
```
Comment (DanieleSalatti) [inline client/src/components/rounds/Apply.tsx:94]: Can you elaborate a bit on this comment/todo?
If I had to address it I wouldn't know what it is referring to. Maybe create an issue and add it to the comment?

**Comment by DanieleSalatti** [comment]
The "X" should stay at the top?

Schermata 2022-09-27 alle 15 46 50

File: client/src/components/rounds/Apply.tsx (hunk 3 above)
Comment (codenamejason) [reply client/src/components/rounds/Apply.tsx:94]: Yeah it seems to redirect before TX is confirmed.

File: client/src/components/grants/Details.tsx (hunk 2 above)
Comment (codenamejason) [reply client/src/components/grants/Details.tsx]: @gravityblast don't we need to update our state object to handle this updated property `roundName` from round-manager? They added the round name last week to the metadata we just need the address to call it 🤔 looking to update this, can we pair tomorrow and tackle this?

Screen Shot 2022-09-27 at 3 38 05 PM

Comment (DanieleSalatti) [reply client/src/components/grants/Details.tsx]: Nah, you can leave it - I'll tackle it as part of #386

File: client/src/components/rounds/Apply.tsx (hunk 3 above)
Comment (DanieleSalatti) [reply client/src/components/rounds/Apply.tsx:95]: I meant if you can add more context to the comment 😛

File: client/src/components/grants/Details.tsx (hunk 1 above)
Comment (willsputra) [reply client/src/components/grants/Details.tsx]: This depends on the round operators I think, not sure if they'll send an email to everyone or just the approved ones. What do you think @michellema1208 ?

Comment (michellema1208) [reply client/src/components/grants/Details.tsx]: Hey @DanieleSalatti @willsputra - yeah let's make this message a bit more generic for now - maybe let's just have the copy say "Thanks for applying to <Program> <Round>! Your application has been received, and the <Program> team will review and reach out with next steps." 

I think showing MVP application status will be a next good-to-have feature we can prioritize, pending completion of priority tickets.

Comment (codenamejason) [reply client/src/components/grants/Details.tsx]: I like this approach @michellema1208 I can update to show  "Thanks for applying to ! Your application has been received, and the team will review and reach out with next steps."

Comment (codenamejason) [reply client/src/components/grants/Details.tsx]: @michellema1208 going to move adding the alert message to another PR to use @gravityblast new component. Just let me know on the copy for now.

**Review by gravityblast — APPROVED** [review]

**Pull request merged**